File: cs50x/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_dynamic_spaces(int row, int height);
void print_dynamic_hashtags(int row);

int main(void)
{
    // Ask user for input
    int height;
    do
    {
        height = get_int("What height do you want the pyramid to be (between 1-8 inclusive)? ");
    }
    while (height < 1 || height > 8);

    for (int i = 1; i <= height; i++)
    {
        // call dynamic spacing function with height and i as inputs
        print_dynamic_spaces(i, height);

        // call dynamic hashtag function with i as input
        print_dynamic_hashtags(i);

        // write two spaces
        printf("  ");

        // call dynamic hashtag function with i as input
        print_dynamic_hashtags(i);

        // write new line
        printf("\n");
    }
}

void print_dynamic_spaces(int row, int height)
{
    for (int i = height - row; i > 0; i--)
    {
        printf(" ");
    }
}

void print_dynamic_hashtags(int row)
{
    for (int i = 0; i < row; i++)
    {
        printf("#");
    }
}

File: cs50x/sentimental-mario-more/mario.py
import cs50

# get input from user
height = cs50.get_int("Height: ")

while (height < 1 or height > 8):
    # reprompt
    height = cs50.get_int("Height: ")

for h in range(1, height + 1):
    # print leading spaces
    for spaces in range(height - h, 0, -1):
        print(" ", end="")

    # print left hashes
    for hashes in range(h):
        print("#", end="")

    # print middle two spaces
    print("  ", end="")

    # print right hashes
    for hashes in range(h):
        print("#", end="")

    # move to next line
    print()
